core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h

```diff
@@ -25,6 +25,23 @@
 #include "vendor/secp256k1-zkp/include/secp256k1_preallocated.h"
 #include "vendor/secp256k1-zkp/include/secp256k1_rangeproof.h"
 #include "vendor/secp256k1-zkp/include/secp256k1_recovery.h"
+#include "vendor/secp256k1-zkp/include/secp256k1_surjectionproof.h"
+
+#define RANGEPROOF_SIGN_BUFFER_SIZE 5134
+
+#if USE_REDUCED_SURJECTION_PROOF_SIZE
+// For firmware (where we have limited stack and heap), should result in <1kB
+// secp256k1_surjectionproof struct.
+#define SURJECTIONPROOF_MAX_USED_INPUTS 16
+#else
+// For emulator - to allocate >8kB secp256k1_surjectionproof struct.
```

Comment:
I'm probably just not seeing the obvious but why do we want different code on the emulator?

Reply:
I wanted to support the verification of surjection proofs on the emulator (since we disabled it on the device, to reduce stack usage at https://github.com/ElementsProject/secp256k1-zkp/pull/57):
https://github.com/romanz/trezor-firmware/pull/1/files#diff-1aaa7015e3c406adc561e688e3404305R199
Not sure that it's the best way though... WDYT?

Reply:
Update: I removed the different definitions between emulator and device, since we now allocate the full surjection proof on CCMRAM (https://github.com/romanz/trezor-firmware/commit/7144d566e31be8986726bd5051ef67ca3db532a8).